File: backend/Api/APOD/GetAPOD.js
const axios = require('axios');

const GetAPOD = async(req, res) =>{
    try {
        const date  = new Date().toISOString().split('T')[0];
        const response = await axios.get('https://api.nasa.gov/planetary/apod', {
          params: { api_key: process.env.NASA_API_KEY, date, thumbs: true },
        });
        res.json(response.data);
      } catch (error) {
        res.status(500).json({ error: 'Failed to fetch APOD' });
      }
}

module.exports = GetAPOD;

File: nasa_ai_agent/api/LLM/Agent_Function/get_apod.py
import requests
from django.conf import settings
import json

def get_apod(date: str = '') -> dict:

    print(f"Calling APOD API with date: {date}")
    try:
        apod_url = "https://api.nasa.gov/planetary/apod"
        params = {"api_key": settings.NASA_API_KEY}
        if date:
            params["date"] = date
        response = requests.get(apod_url, params=params)
        response.raise_for_status()
        apod_data = response.json()
        
        # Simplify APOD data
        result = {
            "date": apod_data.get("date", ""),
            "title": apod_data.get("title", ""),
            "explanation": apod_data.get("explanation", ""),
            "url": apod_data.get("url", ""),
            "media_type": apod_data.get("media_type", "")
        }
        return {"status": "success", "data": result, "error": None}
    except requests.RequestException as e:
        return {"status": "error", "data": None, "error": str(e)}
    except Exception as e:
        return {"status": "error", "data": None, "error": str(e)}